viite-UI/src/model/RoadAddressProjectCollection.js

```diff
@@ -2,25 +2,110 @@
-      var dataJson = {id: 0, status:1 ,name : data[0].value, startDate: data[1].value , additionalInfo :  data[2].value, roadNumber : data[3].value === '' ? 0 : parseInt(data[3].value), startPart: data[4].value === '' ? 0 : parseInt(data[4].value), endPart : data[5].value === '' ? 0 : parseInt(data[5].value) };
+      var projectid=0;
+      if (projectinfo!=null)
+      {
+        projectid=projectinfo.id;
+      }
+      var dataJson = {id: projectid, status:1 ,name : data[0].value, startDate: data[1].value , additionalInfo :  data[2].value,roadpartlist: dirtyRoadSegmentLst};
+
       backend.createRoadAddressProject(dataJson, function(result) {
-        eventbus.trigger('roadAddress:projectSaved', result);
+        console.log(result.success);
+        if(result.success === "ok") {
+          projectinfo={id:result.project.id, additionalInfo:result.project.additionalInfo, status:result.project.status, startDate:result.project.startDate};
+          eventbus.trigger('roadAddress:projectSaved', result);
+        dirtyRoadSegmentLst=[];
+        }
+        else {
+          eventbus.trigger('roadAddress:projectValidationFailed', result);
+        }
       }, function() {
         eventbus.trigger('roadAddress:projectFailed');
       });
     };
 
+      var addSmallLabel = function(label){
+          return '<label class="control-label-small">'+label+'</label>';
+      };
+
+      var updateforminfo = function(formInfo)
+      {
+          $("#roadpartList").html(formInfo)
+      };
+
+      var parseroadpartinfoToresultRow = function() {
+      var listContent='';
+      _.each(currentRoadSegmentList, function(row) {
+        listContent+= addSmallLabel(row.roadNumber)+ addSmallLabel(row.roadPart)+ addSmallLabel(row.lenght)+ addSmallLabel(row.discontinuity)+ addSmallLabel(row.ely) +
+            '</div>';
+          }
+
+      );
+        return listContent
+      };
+
+
+      var addToCurrentRoadPartList = function(querryresult) {
+          var qRoadparts=[];
+          _.each(querryresult.roadparts, function(row) {
+              qRoadparts.push(row)
+          });
+
+          var sameElements=objectIntersection(qRoadparts,currentRoadSegmentList,function(arrayarow,arraybrow){return arrayarow.roadpartid===arraybrow.roadpartid; });
+          _.each(sameElements, function(row)
+          {
+              _.remove(qRoadparts,row)
+          });
+          _.each(qRoadparts,function (row){
+              currentRoadSegmentList.push(row);
+              dirtyRoadSegmentLst.push(row);
+          });
+      };
+
+
+      function objectIntersection(a, b, areEqualFunction) {
+          var results = [];
+
+          for(var i = 0; i < a.length; i++) {
+              var aElement = a[i];
+              var existsInB = _.any(b, function(bElement) { return areEqualFunction(bElement, aElement); });
+
+              if(existsInB) {
+                  results.push(aElement);
+              }
+          }
+          return results;
+      }
```

Comment:
` function objectIntersection(a, b, areEqualFunction) { `
Maybe is better to give more intuitive.